dspace-api/src/main/java/org/dspace/importer/external/liveimportclient/service/LiveImportClientImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.importer.external.liveimportclient.service;
+
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.config.RequestConfig.Builder;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of {@link LiveImportClient}.
+ *
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk at 4science dot com)
+ */
+public class LiveImportClientImpl implements LiveImportClient {
+
+    private final static Logger log = LogManager.getLogger();
+
+    private CloseableHttpClient httpClient;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Override
+    public String executeHttpGetRequest(int timeout, String URL, Map<String, String> requestParams) {
```

Comment:
Again, if we are starting with getting this PR ready **first**, then I think this method needs to have direct tests.  It's extremely important to all your import sources, but it doesn't have its own tests (that I can see).

Reply:
@tdonohue thanks for the feedback, unfortunately the original estimate did not include testing for this part, if you agree, we found this guide https://www.baeldung.com/spring-mocking-webclient#mockwebserver that allows testing this part well, but it takes another 5-6 hours of work.

Reply:
@Micheleboychuk : Let's just leave this as-is for now. I appreciate your clear response that this would require a mock web server to fully test.  Looking at other code in our codebase, it looks like we don't currently require mock web server tests elsewhere.  So, I'll just withdraw this request & we can depend on manual testing instead.